### PR TITLE
Write steps/environment to artifacts folder

### PR DESCRIPTION
**Pull request by nkatzman**
This PR adds in a helper function called `writeArtifact` that can be used to write objects to disk.

It's being used to write the screwdriver job steps and environment to the artifacts directory in a json file

### CODE REVIEW

**Comment by jer** [comment]
👍

File: launch.go [hunk 1]
```diff
@@ -96,6 +100,21 @@ func createWorkspace(rootDir string, srcPaths ...string) (Workspace, error) {
 	return w, nil
 }
 
+func writeArtifact(aDir string, fName string, artifact interface{}) error {
+	data, err := json.MarshalIndent(artifact, "", strings.Repeat(" ", 4))
+	if err != nil {
+		return fmt.Errorf("mashaling artifact: %v ", err)
```
Comment (tkyi) [inline launch.go]: marshalling artifact*

Comment (jer) [inline launch.go]: From https://en.wikipedia.org/wiki/Marshalling_(computer_science):

```
In computer science, marshalling or marshaling is the process of transforming the memory representation of an object to a data format suitable for storage or transmission
```

Comment (nkatzman) [inline launch.go]: well that worked out to my benefit

Comment (tkyi) [inline launch.go]: yeah but Noah is at least missing an `r`

Comment (nkatzman) [inline launch.go]: true..

Comment (tkyi) [inline launch.go]: typical..

Comment (jer) [inline launch.go]: Oof. Letters. Am I rigt?

**Comment by tkyi** [comment]
👍  seems okay

**Pull request merged**